eip712-adapter-moshi/src/main/java/pm/gnosis/eip712/adapters/moshi/TypeAdapters.kt

```diff
@@ -0,0 +1,75 @@
+package pm.gnosis.eip712.adapters.moshi
+
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.JsonQualifier
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.ToJson
+import pm.gnosis.crypto.utils.asEthereumAddressChecksumString
+import pm.gnosis.model.Solidity
+import pm.gnosis.models.Wei
+import pm.gnosis.utils.*
+import java.math.BigInteger
+
+object MoshiBuilderFactory {
+    fun makeMoshiBuilder(): Moshi.Builder {
+        return Moshi.Builder()
+            .add(WeiAdapter())
+            .add(HexNumberAdapter())
+            .add(DecimalNumberAdapter())
+            .add(DefaultNumberAdapter())
+            .add(SolidityAddressAdapter())
+    }
+}
+
+class WeiAdapter {
+    @ToJson
+    fun toJson(wei: Wei): String =
+        StringBuilder("0x").append(wei.value.toString(16)).toString()
+
+    @FromJson
+    fun fromJson(wei: String): Wei {
+        return Wei(wei.parseToBigInteger())
+    }
+}
+
+class HexNumberAdapter {
+    @ToJson
+    fun toJson(@HexNumber hexNumber: BigInteger): String = hexNumber.toHexString()
+
+    @FromJson
+    @HexNumber
+    fun fromJson(hexNumber: String): BigInteger = hexNumber.hexAsBigInteger()
+}
+
+class DecimalNumberAdapter {
+    @ToJson
+    fun toJson(@DecimalNumber bigInteger: BigInteger): String = bigInteger.toString()
+
+    @FromJson
+    @DecimalNumber
+    fun fromJson(decimalNumber: String): BigInteger = decimalNumber.toBigInteger()
+}
+
+class DefaultNumberAdapter {
```

Comment:
I would rename this one to HexNunberAdapter

Reply:
Ohhh we have that already, maxbe then this one should try to detect if it is a decimal or hex number based on the prefix

Reply:
But not sure how to handle the "toJson" in that case

Reply:
Maybe toJson should default to the hex representation (it is the default for most ethereum related logic)

Reply:
Then we can remove it. But I think this is supposed to be the fallback if the passed number is not annotated as @HexNumber.

Reply:
Yes saw this too, as said maybe keep it and use the parseToBigInteger function in the fromJson part

Reply:
Fixed in 6368c01e9b0d89219f2cfef54a7d13dc3a62e2c8